cmd/wallet/main.go

```diff
@@ -1,163 +1,391 @@
+			return nil, errors.Wrap(wrongProgramArguments, "no seed phrase was provided")
+		}
+		if opts.IsSeedPhraseBase58 {
+			return nil, errors.Wrap(wrongProgramArguments, "seed phrase base-58-encoding flag was true, but non-base-58 option was chosen")
+		}
+
+		accountSeed, pk, sk, address, err := generateOnSeedPhrase(opts.SeedPhrase, opts.AccountNumber, opts.Scheme)
+		if err != nil {
+			return nil, err
+		}
+		walletCredentials = &WalletCredentials{
+			accountSeed: accountSeed,
+			pk:          pk,
+			sk:          sk,
+			address:     address,
+		}
+	case seedPhraseBase58Opt:
+		if !opts.IsSeedPhraseBase58 {
+			return nil, errors.Wrap(wrongProgramArguments, "seed phrase base-58-encoding flag was false, but base-58 option was chosen")
+		}
+		b, err := base58.Decode(opts.SeedPhrase)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to decode base58-encoded seed phrase")
+		}
+		decodedSeedPhrase := string(b)
+
+		accountSeed, pk, sk, address, err := generateOnSeedPhrase(decodedSeedPhrase, opts.AccountNumber, opts.Scheme)
+		if err != nil {
+			return nil, err
+		}
+		walletCredentials = &WalletCredentials{
+			accountSeed: accountSeed,
+			pk:          pk,
+			sk:          sk,
+			address:     address,
+		}
+	case accountSeedBase58Opt:
+		if opts.Base58AccountSeed == "" {
+			return nil, errors.Wrap(wrongProgramArguments, "no base58 account seed was provided")
+		}
+		accountSeed, err := crypto.NewDigestFromBase58(opts.Base58AccountSeed)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to decode base58-encoded account seed")
+		}
+		pk, sk, address, err := generateOnAccountSeed(accountSeed, opts.Scheme)
+		if err != nil {
+			return nil, err
+		}
+		walletCredentials = &WalletCredentials{
+			accountSeed: accountSeed,
+			pk:          pk,
+			sk:          sk,
+			address:     address,
+		}
+
+	default:
+		showUsageAndExit()
 	}
 
-	fmt.Print("Enter seed: ")
-	seed, err := gopass.GetPasswd()
+	return walletCredentials, nil
+}
+
+func createWallet(opts Opts, command string) error {
+	walletPath, err := getWalletPath(opts.PathToWallet)
 	if err != nil {
-		fmt.Println("Interrupt")
-		return
+		return errors.Wrap(err, "failed to handle wallet's path")
 	}
 
-	if opts.Base58 {
-		seed, err = base58.Decode(string(seed))
+	var walletCredentials *WalletCredentials
+
+	walletCredentials, err = generateWalletCredentials(command, opts)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate wallet's credentials")
+	}
+	if walletCredentials == nil {
+		return errors.New("failed to generate wallet's credentials")
+	}
+
+	fmt.Print("Enter password to encode your account seed: ")
+	pass, err := gopass.GetPasswd()
+	if err != nil {
+		return errors.Wrap(err, "failed to get the password to encode the account seed")
+	}
+
+	if len(pass) == 0 {
+		return errors.Wrap(err, "the password's length is zero")
+	}
+
+	var wlt wallet.Wallet
+	if exists(walletPath) {
+		fmt.Print("Wallet already exists on the provided path. Rewrite? Y/N: ")
```

Comment:
This routine doesn't allow to add new seed to existing wallet. It only allows to overwrite it. The question should be "Wallet already exist. Do you want to add seed to existing wallet or overwrite it (existing seeds will be lost)? [A]dd/[O]verwrite/[C]ancel".